spec/meilisearch/index/settings_spec.rb

```diff
@@ -680,4 +682,51 @@ def update_synonyms(index, synonyms)
 
     client.wait_for_task(task['taskUid'])
   end
+
+  context 'On pagination sub-routes' do
+    let(:index) { client.index(uid) }
+    let(:pagination) { { maxTotalHits: 3141 } }
+
+    before { client.create_index!(uid) }
+
+    it 'gets default values of pagination' do
+      settings = index.pagination.transform_keys(&:to_sym)
+      expect(settings).to be_a(Hash)
+      expect(settings).to eq(default_pagination)
+    end
+
+    it 'updates pagination' do
+      task = index.update_pagination(pagination)
+      client.wait_for_task(task['taskUid'])
+      expect(task['type']).to eq('settingsUpdate')
+      expect(index.pagination.transform_keys(&:to_sym)).to eq(pagination)
+    end
+
+    it 'updates pagination at null' do
+      task = index.update_pagination(pagination)
+      expect(task['type']).to eq('settingsUpdate')
+      client.wait_for_task(task['taskUid'])
+
+      task = index.update_pagination(nil)
+
+      expect(task['type']).to eq('settingsUpdate')
+      client.wait_for_task(task['taskUid'])
+
+      expect(index.pagination.transform_keys(&:to_sym)).to eq(default_pagination)
+    end
+
+    it 'resets pagination' do
+      task = index.update_pagination(pagination)
+      expect(task['type']).to eq('settingsUpdate')
+      client.wait_for_task(task['taskUid'])
+
+      task = index.reset_pagination
+
+      expect(task['type']).to eq('settingsUpdate')
+      client.wait_for_task(task['taskUid'])
+
+      expect(index.task(task['taskUid'])['status']).to eq('succeeded')
+      expect(index.pagination.transform_keys(&:to_sym)).to eq(default_pagination)
+    end
```

Comment:
No need to add the check of the task's status since the `wait_for_task` will take care of that for you.

```suggestion
    it 'resets pagination' do
      task = index.update_pagination(pagination)
      client.wait_for_task(task['taskUid'])

      task = index.reset_pagination
      client.wait_for_task(task['taskUid'])

      expect(index.pagination.transform_keys(&:to_sym)).to eq(default_pagination)
    end
```